core/lib/zksync_core/src/state_keeper/io/mod.rs

```diff
@@ -59,12 +59,12 @@ pub struct MiniblockParams {
 }
 
 /// Parameters for a new L1 batch returned by [`StateKeeperIO::wait_for_new_batch_params()`].
-#[derive(Debug)]
-pub struct L1BatchParams {
+#[derive(Debug, Clone)]
+pub struct L1BatchParams<H = H256> {
     /// Protocol version for the new L1 batch.
     pub(crate) protocol_version: ProtocolVersionId,
-    /// State hash of the previous L1 batch.
-    pub(crate) previous_batch_hash: H256,
+    /// State hash of the previous L1 batch. `()` means that it's not necessarily known at this point.
+    pub(crate) previous_batch_hash: H,
```

Comment:
Subjective -1 on this one. It seems to be a fancy way to avoid using `Option` here, but at the same time, this approach isn't common in the workspace. Given that the level of Rust expertise varies throughout the team significantly, I'm afraid that it may be confusing to some people, so personally I'd stick with a simpler approach.